hooks/checkout

```diff
@@ -1,9 +1,14 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-set -euo pipefail
+set -eo pipefail
 
 if [[ "$BUILDKITE_PLUGIN_SMOOTH_CHECKOUT_SKIP_CHECKOUT" == "true" ]]; then
     echo "--- :fast_forward: skipping checkout"
     echo "Because 'skip_checkout' configuration was set as true in pipeline YAML"
     exit 0
-fi
\ No newline at end of file
+fi
+
+CHECKOUT_DIR="$HOME/buildkite-checkouts/$BUILDKITE_BUILD_ID/$BUILDKITE_JOB_ID"
+echo "Creating checkout directory $CHECKOUT_DIR ..."
```

Comment:
```
echo "--- :open_file_folder: setting up checkout directory"
echo "Creating checkout directory: $CHECKOUT_DIR" 
```

Reply:
Maybe we should refer to `checkout directory` as `workspace` instead. Because the directory in which the checkout happens will depend on the clone URL.